### PR TITLE
Add AuthenticationType enum to typings

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Got annoyed at the various copies of AuthenticationType values being duplicated everywhere - and this is something that extensions would find very useful when creating or using connection infos (especially since the values aren't really documented anywhere).

Purposely keeping it pretty generic though as just a list of well-known auth types since we don't really have a good way of telling the user what auth types a specific provider may or may not support. If that was something that ended up being useful I think adding functionality to the providers to say what auth types they support would be good. 

This does mean we need to be extra careful when changing these values though - since they're runtime objects if we were to change the strings at all that could break a lot of stuff (but that's pretty much the same as what would happen now with changing the hardcoded string constants so it's not really that much different. The main difference here is that extensions using this would also then have their values changed which could be unexpected).

### CODE REVIEW

File: extensions/sql-database-projects/src/models/dataSources/sqlConnectionStringSource.ts [hunk 1]
```diff
@@ -44,11 +45,11 @@ export class SqlConnectionDataSource extends DataSource {
...
 	public get authType(): string {
 		if (this.azureMFA) {
-			return constants.azureMfaAuth;
+			return vscodeMssql.AuthenticationType.AzureMFA;
 		} else if (this.integratedSecurity) {
-			return constants.integratedAuth;
+			return vscodeMssql.AuthenticationType.Integrated;
 		} else {
-			return constants.sqlAuth;
+			return 'SqlAuth';
```
Comment (Charles-Gagnon) [inline extensions/sql-database-projects/src/models/dataSources/sqlConnectionStringSource.ts:52]: @kisantia Is this value supposed to be different than SqlLogin?

File: src/sql/azdata.proposed.d.ts [hunk 2]
```diff
@@ -403,11 +403,44 @@ declare module 'azdata' {
 	 * Add optional azureAccount for connectionWidget.
 	 */
```
Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts:404]: please fix the comment while you are working on this.

File: src/sql/azdata.proposed.d.ts [hunk 3]
```diff
@@ -403,11 +403,44 @@ declare module 'azdata' {
...
 
+	export namespace connection {
+		/**
+		 * Well-known Authentication types commonly supported by connection providers.
+		 */
+		export enum AuthenticationType {
+			/**
+			 * Username and password
+			 */
+			SqlLogin = 'SqlLogin',
```
Comment (alanrenmsft) [inline src/sql/azdata.proposed.d.ts:420]: I always wanted to change this to UsernamePassword 😄.

**Review by alanrenmsft — APPROVED** [review]

File: extensions/azurecore/src/azureResource/providers/databaseServer/databaseServerTreeDataProvider.ts [hunk 4]
```diff
@@ -3,7 +3,7 @@
  *  Licensed under the Source EULA. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-import { ExtensionNodeType, TreeItem } from 'azdata';
+import { ExtensionNodeType, TreeItem, connection } from 'azdata';
```
Comment (cheenamalhotra) [inline extensions/azurecore/src/azureResource/providers/databaseServer/databaseServerTreeDataProvider.ts:6]: This can be removed from here.

**Review by cheenamalhotra — APPROVED** [review]

File: extensions/azurecore/src/azureResource/providers/databaseServer/databaseServerTreeDataProvider.ts (hunk 4 above)
Comment (Charles-Gagnon) [reply extensions/azurecore/src/azureResource/providers/databaseServer/databaseServerTreeDataProvider.ts:6]: Not just a nit - compilation failing problem 😄

**Pull request merged**